src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable react/no-did-update-set-state */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import './Pagination.scss';
+
+const CN = 'pagination';
+
+class Pagination extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isValidInput: true,
+      isAnimation: false
+    };
+
+    this.setPage = this.setPage.bind(this);
+    this.renderNumbers = this.renderNumbers.bind(this);
+    this.goToPrevPage = this.goToPrevPage.bind(this);
+    this.goToNextPage = this.goToNextPage.bind(this);
+    this.checkThenChangePage = this.checkThenChangePage.bind(this);
+    this.onAnimationEnd = this.onAnimationEnd.bind(this);
+    this.inputRef = React.createRef();
+  }
+
+  componentDidMount() {
+    const { currentPage } = this.props;
+
+    this.inputRef.current.value = currentPage;
+  }
+
+  componentDidUpdate({ currentPage: prevCurrentPage }) {
+    const { currentPage, isValidInput, numberOfPages } = this.props;
+
+    if (prevCurrentPage !== currentPage) {
+      this.inputRef.current.value = currentPage;
+
+      !isValidInput
+        && this.setState({
+          isValidInput: true
+        });
+    }
+
+    if (numberOfPages === 1) {
+      this.inputRef.current.value = 1;
+    }
+  }
+
+  onAnimationEnd() {
+    this.setState({
+      isAnimation: false
+    });
+  }
+
+  setCurrentPage(value) {
+    const { setCurrentPage } = this.props;
+
+    setCurrentPage(+value);
+  }
+
+  setPage(e) {
+    this.setCurrentPage(+e.target.innerText);
+  }
+
+  goToPrevPage() {
+    const { currentPage } = this.props;
+
+    if (currentPage !== 1) {
```

Comment:
If you disable go to previous page button, do you really need this check here?

Reply:
I got it! will fix!